File: monte-carlo-pi/pi.py
from random import *
from math import sqrt

max = 10000
inside = 0
for n in xrange(max):
    x, y = random(), random()
    print sqrt(x*x + y*y)
    if sqrt(x*x + y*y) <= 1:
        inside += 1

print(inside * 4.0 / max)

File: monte-carlo-pi/pi.rb
require 'securerandom'

N = 10_000
inside = N.times.select do
  x = SecureRandom.random_number
  y = SecureRandom.random_number
  Math.sqrt(x*x + y*y) < 1.0
end.size

p inside * 4.0 / N





